### DISCUSSION TITLE
Why is Member null when using Guild.getMemberById(...)?

### DISCUSSION THREAD

**Discussion by DomiWe** [discord-jda/JDA#1599] [Questions and Help] [answered]
jda.getGuildById(...).getMemberById(...)

I get null, but on the Server in my Guild the Member exists.

![DEBUG_VARIABLES](https://user-images.githubusercontent.com/37793935/114949406-44dff680-9e51-11eb-895a-38e5a81174eb.PNG)
![DEBUG_CODE](https://user-images.githubusercontent.com/37793935/114949410-46112380-9e51-11eb-9151-cec4e872e2a8.PNG)

**Comment by RedDaedalus** [comment, marked as answer]
When using methods such as `getMember()` or `getMemberById(String)` there's a possibility that the returned Member is null.
There can be many reasons, but most of the time the cause is that a Member isn't cached (yet).

When using `createLight(String)` or `createDefault(String)` for building a JDA/ShardManager instance JDA will disable Member Caching and Member Chunking.
This has the side-effect that JDA doesn't "know" a member yet and therefore returns null since a Member is nullable.

There are several options that you can do instead:
- If you get the member of a Message-based event such as GuildMessageReceivedEvent, use `getAuthor()` instead as this returns a User object which won't be null.
- Use `retrieveMember()`. This is a RestAction, so don't forget to `queue()` it!
- Have Member Caching and Member Chunking active. This requires the priviledged `GUILD_MEMBER` intend to be enabled.

Please read this page for more info:
https://github.com/DV8FromTheWorld/JDA/wiki/Gateway-Intents-and-Member-Cache-Policy
